### PR TITLE
Initial jdk18 files

### PR DESCRIPTION
**Pull request by smlambert**
Signed-off-by: Shelley Lambert <slambert@gmail.com>

### CODE REVIEW

**Comment by smlambert** [comment]
@sophia-guo - can you give me guidance or point me at doc on what is needed for adding a new jdk version for the Linux installers (and how to test it)?

**Comment by sophia-guo** [comment]
We might not have the doc on how to contribute it :-(. For testing in local the [README](https://github.com/adoptium/installer/blob/master/linuxNew/README.md) should help. Jenkins job you can use https://ci.adoptopenjdk.net/view/work-in-progress/job/Sophia-adoptium-packages-linux-pipeline/

**Comment by sophia-guo** [comment]
I think your change for Debian should be good. You will also need to add 18 to redhat and suse, similar to other version https://github.com/adoptium/installer/blob/master/linuxNew/jdk/redhat/src/main/packaging/temurin/17/temurin-17-jdk.spec .
https://github.com/adoptium/installer/blob/master/linuxNew/jdk/suse/src/main/packaging/temurin/17/temurin-17-jdk.spec

**Comment by sophia-guo** [comment]
To support jdk18 in jenkins job [jenkinsfile](https://github.com/adoptium/installer/blob/master/linuxNew/Jenkinsfile) will also need to update.

**Comment by smlambert** [comment]
I've pushed some of the changes, but leaving in draft, as we did not produce JREs, so presume will have to edit the spec files to reflect that.

File: linuxNew/jdk/redhat/src/main/packaging/temurin/18/temurin-18-jdk.spec [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    update-alternatives --remove javac %{prefix}/bin/javac
+fi
+
+%files
+%defattr(-,root,root)
+%{prefix}
+/usr/lib/tmpfiles.d/%{name}.conf
+
+%changelog
+* Tue Apr 1 2022 Eclipse Adoptium Package Maintainers <temurin-dev@eclipse.org> 18.0.0.0.0.36-1.adopt0
```
Comment (karianna) [inline linuxNew/jdk/redhat/src/main/packaging/temurin/18/temurin-18-jdk.spec:260]:
```suggestion
* Fri Apr 1 2022 Eclipse Adoptium Package Maintainers <temurin-dev@eclipse.org> 18.0.0.0.0.36-1.adopt0
```

**Review by karianna — COMMENTED** [review]
Made some suggestions

**Review by jerboaa — APPROVED** [review]
Spec files look fine to me.

**Comment by gdams** [comment]
@smlambert you'll also need to add JDK18 to the GitHub actions workflow here https://github.com/adoptium/installer/blob/master/.github/workflows/linux.yml

**Review by sophia-guo — APPROVED** [review]
LGTM.

**Pull request merged**